File: 2418. Sort the People/solution.cpp
class Solution {
public:
  vector<string> sortPeople(vector<string>& names, vector<int>& heights) {
    unordered_map<int, string> dictionary;
    for (int index = 0; index < names.size(); index++){
      string name = names[index];
      int height = heights[index];
      dictionary[height] = name;
    }

    sort(heights.begin(), heights.end());

    vector<string> result;
    for (int index = heights.size() - 1; index > -1; index--){
      int height = heights[index];
      result.push_back(dictionary[height]);
    }

    return result;
  }
};

File: 2418. Sort the People/solution.java
class Solution {
  public String[] sortPeople(String[] names, int[] heights) {
    Map<Integer, String> dictionary = new HashMap<>();
    for (int index = 0; index < names.length; index++){
      String name = names[index];
      int height = heights[index];
      dictionary.put(height, name);
    }

    Arrays.sort(heights);

    String[] result = new String[names.length];
    for (int index = heights.length - 1; index > -1; index--){
      int height = heights[index];
      result[heights.length - index - 1] = dictionary.get(height);
    }

    return result;
  }
}